### ISSUE TITLE
feature request: a matomo like setup.php installation

### ISSUE THREAD

**Issue by supervisitor** [plausible/hosting#15]
a simpler, matomo like installation process:

- unpack the downloaded zip-file on the server
- open the file setup.php in browser
- make the entries for database (mysql) and first admin user

finished... go and integrate the js into all webs...

**Comment by ukutaht** [comment]
Sounds good. A simple starting point would be a flow to add the first admin account through the web UI so you don't have to add admin credentials to the config file.

**Comment by minamorcose** [comment]
yes i was searching for alternative soultion too it would be good to support that method

**Comment by ukutaht** [comment]
I went through installing Sentry just recently and I have to say the install script is very nice. Something to take inspiration from when building our own bootstrap script:

https://github.com/getsentry/onpremise/blob/master/install.sh

**Comment by ukutaht** [comment]
Also very nice configuration screen when booting up for the first time. Most config is done in the UI, only basic system stuff is in the original install script.

![Screenshot from 2021-03-03 15-44-28@2x](https://user-images.githubusercontent.com/3731516/109814741-874dcb00-7c37-11eb-9f85-681bc9780331.png)

![Screenshot from 2021-03-03 15-45-10@2x](https://user-images.githubusercontent.com/3731516/109814749-8a48bb80-7c37-11eb-8cdb-24fb8b3747e1.png)

**Comment by dodedodo** [comment]
plausible analytics is not a php app. You cannot install it as such.

**Comment by ukutaht** [comment]
We could still provide a script that makes the installation process easier. Doesn't have to be php

**Comment by Philipinho** [comment]
> We could still provide a script that makes the installation process easier. Doesn't have to be php

My thoughts exactly. The PHP process is more of a reference.
I might work out something soon with bash script in my free time.

**Comment by dodedodo** [comment]
There's a `docker-compose.yml` file in this repo. Grab a vps with docker installed, `git clone` this repository and run `docker-compose up -d`. That's 2 commands to get a working installation. It doesn't get much simpler than that in my opinion.

> *I do not recommend self-hosting important/production services if you don't know what you're doing.*

This project works with NodeJS, ClickhouseDB and PostgreSQL. It's simply not the same technology stack as your regular cPanel webhost. It's incompatible. You could probably upload the `docker-compose.yml` to something like heroku or AWS app runner, just like you would a setup.php to cPanel. I have zero experience with these services so I cannot help you.

**Comment by minamorcose** [comment]
> There's a `docker-compose.yml` file in this repo. Grab a vps with docker installed, `git clone` this repository and run `docker-compose up -d`. That's 2 commands to get a working installation. It doesn't get much simpler than that in my opinion.
> 
> > _I do not recommend self-hosting important/production services if you don't know what you're doing._
> 
> This project works with NodeJS, ClickhouseDB and PostgreSQL. It's simply not the same technology stack as your regular cPanel webhost. It's incompatible. You could probably upload the `docker-compose.yml` to something like heroku or AWS app runner, just like you would a setup.php to cPanel. I have zero experience with these services so I cannot help you.

i would perfer some bash script because not all host support docker spicially Paas Services like Jelastic ... you don't need to support all the platform it enoghe to support Debian and maybe RHEL
